### PR TITLE
MSEARCH-127: Describe on-premise ES configuration in readme.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
Add readme about on-premise ES config and recommended production configuration.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +57,48 @@ These languages will be added on tenant init and applied to index. Example usage
...
+
+For production installations it is strongly recommended enabling security for instance and set-up user/password. The user
+must have at least following permissions:
+* Create/delete/update index and mappings for it.
+* Create/update/delete documents in index.
+
+### Recommended production set-up
+
+The data nodes Elasticsearch configuration completely depends on the data.
+If there are 7 mln of instances the configuration with 2 nodes with 8Gb RAM and 500 Gb disk (AWS m5.large) works well.
```
Comment (hjiebsco) [inline README.md:94]: Does the 2 data nodes work with production setup (dedicated master node)? The default index config has `"number_of_replicas": 2`. See https://github.com/folio-org/mod-search/blob/master/src/main/resources/elasticsearch/index/instance.json#L4.

Comment (bohdan-suprun) [reply README.md:94]: I quoted @MikhailFokanov's reply to John in slack, @MikhailFokanov could you clarify?

Comment (predictron-cloud) [reply README.md:94]: @hjiebsco, @bohdan-suprun I wrote that 2 nodes, which both are master and data is the most basic configuration. Also I wrote, that for fault tolerance Elastic needs dedicated master nodes not to have split brain. For full fault tolerance there should be 3 data nodes to have ability to write and read to index even if one node is lost. But maybe guys should consider tradeoff between full fault tolerance and budget for Elastic AWS hosting.

My comment was:
|The data nodes Elasticsearch configuration completely depends on the data. If there are 7 mln of instances the configuration with 2 nodes with 8Gb RAM and 500 Gb disk (AWS m5.large) works well. We performed performance tests for this configuration and it showed good results. But I would recommend to performing additional performance testing (try to reindex and search with different configurations) with different type of nodes, and see what configuration is sufficient for what data volume. **Also for fault tolerance Elastic needs dedicated master nodes (not to have quorum problem which is called split brain) with less powerful configuration (see https://www.elastic.co/guide/en/cloud-enterprise/current/ece-ha.html).**
This described suggestion **the most basic configuration**, but having more expensive configuration could be money wasting if the basic one works well for the data.

Comment (MikhailFokanov) [reply README.md:94]: @hjiebsco, @bohdan-suprun I wrote that 2 nodes, which both are master and data is the most basic configuration. Also I wrote, that for fault tolerance Elastic needs dedicated master nodes not to have split brain. For full fault tolerance there should be 3 data nodes to have ability to write and read to index even if one node is lost. But maybe guys should consider tradeoff between full fault tolerance and budget for Elastic AWS hosting.

My comment was:
The data nodes Elasticsearch configuration completely depends on the data. If there are 7 mln of instances the configuration with 2 nodes with 8Gb RAM and 500 Gb disk (AWS m5.large) works well. We performed performance tests for this configuration and it showed good results. But I would recommend to performing additional performance testing (try to reindex and search with different configurations) with different type of nodes, and see what configuration is sufficient for what data volume. Also for fault tolerance Elastic needs dedicated master nodes (not to have quorum problem which is called split brain) with less powerful configuration (see https://www.elastic.co/guide/en/cloud-enterprise/current/ece-ha.html).
This described suggestion the most basic configuration, but having more expensive configuration could be money wasting if the basic one works well for the data.

Comment (hjiebsco) [reply README.md:94]: > @hjiebsco, @bohdan-suprun I wrote that 2 nodes, which both are master and data is the most basic configuration. Also I wrote, that for fault tolerance Elastic needs dedicated master nodes not to have split brain. For full fault tolerance there should be 3 data nodes to have ability to write and read to index even if one node is lost. But maybe guys should consider tradeoff between full fault tolerance and budget for Elastic AWS hosting.
> 
> My comment was:
> The data nodes Elasticsearch configuration completely depends on the data. If there are 7 mln of instances the configuration with 2 nodes with 8Gb RAM and 500 Gb disk (AWS m5.large) works well. We performed performance tests for this configuration and it showed good results. But I would recommend to performing additional performance testing (try to reindex and search with different configurations) with different type of nodes, and see what configuration is sufficient for what data volume. Also for fault tolerance Elastic needs dedicated master nodes (not to have quorum problem which is called split brain) with less powerful configuration (see https://www.elastic.co/guide/en/cloud-enterprise/current/ece-ha.html).
> This described suggestion the most basic configuration, but having more expensive configuration could be money wasting if the basic one works well for the data.

Consider to mention it explicitly that the 2 nodes performance test was done in a mixed master/data nodes mode.

**Review by hjiebsco — APPROVED** [review]

**Review by pfilippovEPAM — APPROVED** [review]
Looks good to me.

**Comment by bohdan-suprun** [comment]
@hjiebsco, thanks. I've mentioned about the mixed mode.

**Pull request merged**